kerastuner/engine/oracle.py

```diff
@@ -142,7 +142,7 @@ def create_trial(self, tuner_id):
 
         trial_id = trial_lib.generate_trial_id()
 
-        if len(self.trials) >= self.max_trials:
+        if self.max_trials and len(self.trials) >= self.max_trials:
```

Comment:
For the Hyperband oracle it doesn't make sense (since a "trial" could run for either 1 epoch or 100 epochs) and makes more sense to specify something like the number of times the whole Hyperband algorithm should repeat (or the total number of epochs the search should use)